client/v2/client.go

```diff
@@ -61,6 +70,9 @@ type Client interface {
 	// Query makes an InfluxDB Query on the database. This will fail if using
 	// the UDP client.
 	Query(q Query) (*Response, error)
+
+	// Close releases any resources a Client may be using.
+	Close() error
```

Comment:
Yeah, let's not force them pass in  a `net.Conn`. It's too much. I'm OK with not having an `Open()`, it's not a blocker. The changes here are a big improvement regardless.